Atangster/13 add search functionality

atangster opened the pull request:
## Pull Request Details

Link to associated issue: https://github.com/anusii/moviestar/issues/13

# Checklist
Complete the check-list below to ensure your branch is ready for PR.

Flutter Style Guide: https://survivor.togaware.com/gnulinux/flutter-style.html

- [x] Screenshots included in linked issue
- [x] Changes adhere to the team style and coding guideline
- [x] No confidential information
- [x] No duplicated content
- [x] No lint check errors related to your changes (make prep or flutter analyze lib)
- [x] Tested on at least one device
  - [x] Windows
  - [x] Linux
- [x] Added 2 reviewers (or 1 for private repositories then they add another)

atangster committed:
add search functionality

atangster committed:
Fix code format

atangster committed:
Fix import order

cdawei reviewed (approved):
The changes look good and the app worked as expected in my test. Thanks @atangster.

anushkavidanage reviewed (approved):
Looks good. Thanks @atangster

atangster committed:
Merge remote-tracking branch 'origin/dev' into atangster/13_search

atangster commented:
> Merge remote-tracking branch 'origin/dev' into atangster/13_search

- [x] Resolve conflict with dev branch

atangster committed:
run dart format

gjwgit commented on lib/screens/search_screen.dart:
```diff
@@ -70,44 +72,210 @@ class _SearchScreenState extends State<SearchScreen> {
   String? _error;
 
-  /// List of movies matching the search query.
+  /// Comprehensive search results categorized by search type.
```
Does this private class ever get documented? If not we should not be using docstring comments `///`.  I ran `dart doc` to check and it seems not.

gjwgit commented on lib/screens/search_screen.dart:
```diff
@@ -70,44 +72,210 @@ class _SearchScreenState extends State<SearchScreen> {
+    }
+  }
+
+  /// Builds a section of search results for a specific category.
```
I don't think private functions should have docstring comments?

gjwgit commented on lib/screens/search_screen.dart:
This file is getting too large. Need to look for opportunities to modularise the code.

gjwgit commented on lib/services/movie_service.dart:
```diff
@@ -111,10 +111,175 @@ class MovieService {
 
+  /// Searches for movies by actor/person name.
+
+  Future<List<Movie>> searchMoviesByActor(String actorName) async {
```
Seems these could be split out into their own file?

gjwgit reviewed (approved):
- [x] Does the SEARCH button (and THEME button) need to be floating. They stick out from amongst the other buttons along the top. Should they all be floating buttons, or all not?

- [x] Comments (and code lines in general) should not beyond column 80.

- [x] When you modify a file please add your name to the author list.

gjwgit committed:
Updates to style and license template to suit doc strings.

atangster committed:
fix pr comments

- make search button and other icons non-floating
- make private class docs double slash
- modularise movie search service

atangster committed:
fix code format

gjwgit commented:
Ready to merge - thanks @atangster
